### PR TITLE
Add cryptoavatars

### PR DESCRIPTION
**Pull request by mavisakalyan**
## Describe your changes
- Crypto avatars list added to the character select box.
- Character select preview box fixed to the bottom so when you scroll through all the lists it does not scroll with it.
- Changed the show arrow on hover for selecting the characters to just css and removed updateArrowPosition to decrease FPS ( No need to run all those functions for just 1 arrow ).
- Fixed the issue where some characters were facing the other way on preview.
- Fixed the app crashing when certain characters were selected ( problem was that not all models have the Head mesh called exactly "Head" so we modified the code a bit. )
- Styled the select boxes for the CryptoAvatar menu to fit the design better.
## What are the steps for a QA tester to test this pull request?
- Load the scene
- Open the character select menu
- And surf through the newly added CryptoAvatar characters
- Use pagination, hover over the characters and see them in the preview box
- Select the characters and playtest.
## Screenshots and/or video

https://user-images.githubusercontent.com/55106546/183263052-12afeccf-24db-45a5-b3d6-e2638aeccb66.mov

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I am not adding any irrelevant code or assets
- [x] I am only including the changes needed to implement the change
- [x] I have playtested and intentionally tried to find error cases but couldn't

### CODE REVIEW

**Comment by avaer** [comment]
Can we get rid of PNG and use SVG?

https://github.com/webaverse/app/pull/3528/files#diff-94bce46600a509ecd3a2f8c05432084ba0e40346ff825454501753c45efdcd8c

File: src/components/general/character-select/cryptoavatars-loader.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+/* eslint-disable */
+// API DOC https://cryptoavatars.io/integrations
+// NFT Filter DTO SCHEMA at the end of the document
+
+// SKIP & LIMIT supported for pagination on URL params
+
+export async function loadCryptoAvatarsCharacters(
+    url = undefined,
+    ownership = undefined,
+    collectionAddress = '0xc1def47cf1e15ee8c2a92f4e0e968372880d18d1',
```
Comment (avaer) [inline src/components/general/character-select/cryptoavatars-loader.js:10]: These should at least be constants in this file.

File: src/components/general/character-select/CharacterSelect.jsx [hunk 2]
```diff
@@ -213,42 +227,20 @@ export const CharacterSelect = () => {
...
-            }
-        }
-        return map;
-    })();
+    const [ npcPlayerCache, setNpcPlayerCache ] = useState(new Map());
+
+    const [caPagination, setCaPagination] = useState({});
+    const [caItemsPerPage, setCaItemsPerPage] = useState(5);
+    const [caCollection, setCaCollection] = useState(
+        '0xc1def47cf1e15ee8c2a92f4e0e968372880d18d1'
```
Comment (avaer) [inline src/components/general/character-select/CharacterSelect.jsx:237]: This should be a constant.

File: src/components/general/character-select/CharacterSelect.jsx [hunk 3]
```diff
@@ -403,17 +386,67 @@ export const CharacterSelect = () => {
         }
     };
 
+    /** ------------------------- CRYPTOAVATARS IMPLEMENTATION ------------------------ */
+    const getCryptoAvatars = async (
```
Comment (avaer) [inline src/components/general/character-select/CharacterSelect.jsx:390]: This should be in a more abstract util, even if it's just a local `util.js`.

File: src/components/general/character-select/CharacterSelect.jsx [hunk 4]
```diff
@@ -134,24 +136,36 @@ const Character = forwardRef(({
...
 
 export const CharacterSelect = () => {
     const { state, setState } = useContext( AppContext );
     const [ highlightCharacter, setHighlightCharacter ] = useState(null);
     const [ selectCharacter, setSelectCharacter ] = useState(null);
-    const [ lastTargetCharacter, setLastTargetCharacter ] = useState(null);
-    const [ abortFn, setAbortFn ] = useState(null);
     const [ arrowPosition, setArrowPosition ] = useState(null);
-    const [ enabled, setEnabled ] = useState(false);
+    const [ arrowPosition2, setArrowPosition2 ] = useState(null);
```
Comment (avaer) [inline src/components/general/character-select/CharacterSelect.jsx:164]: `arrowPosition2` is not used?

File: src/LightArrow.jsx [hunk 5]
```diff
@@ -246,15 +247,15 @@ export const LightArrow = function({
...
+        enabled ? styles.enabled : styles.enabled,
         up ? styles.up : null,
         down ? styles.down : null,
         className,
       )}
       style={{
-        transform: `translate(${x}px, ${y}px)`,
-        left: ax,
-        top: ay,
+        left: `75%`,
```
Comment (avaer) [inline src/LightArrow.jsx:256]: This could be in the stylesheet. Also, would prefer to use pixel values rather than percentages.

**Comment by avaer** [comment]
This edits `package-lock.json` and `totum`, which I don't think it should.

**Review by avaer — CHANGES_REQUESTED** [review]
Overall looks good, but needs more abstraction and refactoring.

**Comment by mavisakalyan** [comment]
All comments on code cleared ready for review.

**Comment by avaer** [comment]
Code can’t be trusted since it included irrelevant security related files.

**Comment by zenkale** [comment]
Issues:
1-The premium avatars are accessible while they shouldn't be:

https://user-images.githubusercontent.com/111510128/188887493-d3f32639-9040-4663-a9ae-eb8c2fe48b33.mp4

2-The previews are going out of bounds of preview area:

https://user-images.githubusercontent.com/111510128/188887949-874a6363-4193-4e17-a737-60ae5f68a2cd.mp4

3-Selecting the options isn't working as expected. For example: choosing the ownership as free use or owned throws these errors, meanwhile the characters in list remain the same.

image

https://user-images.githubusercontent.com/111510128/188888988-4ee8e666-9fb1-4b43-8896-f3b867549288.mp4

4-Selecting the 25 avatars per page option only gets 20 avatars instead of 25:

image

5-If we select 10, 25 characters per page, the characters appear above the preview of the selected character. Can we fix it so that preview isn't covered?

image

6-When we have the openai key set up, some of the characters play the same intro as we have for above upstreet avatars:

https://user-images.githubusercontent.com/111510128/188890441-d934ebe7-20ad-421d-baf5-2296bfa4f5d8.mp4

**Comment by luixxma** [comment]
Hi, i am going to update this PR with the new nft collection system of our API and fix the problem with premium avatars wich can be selected.

**Comment by lalalune** [comment]
> Hi, i am going to update this PR with the new nft collection system of our API and fix the problem with premium avatars wich can be selected.

Great. We have the same work earmarked, if you can get to it first, great.

We have this merged and in testing over at: https://github.com/webaverse-studios/app

This is kind of our staging server atm

Gonna close this PR in place of the one we have their.

**Comment by lalalune** [comment]
> Hi, i am going to update this PR with the new nft collection system of our API and fix the problem with premium avatars wich can be selected.

Great. We have the same work earmarked, if you can get to it first, great.

We have this merged and in testing over at: https://github.com/webaverse-studios/app

This is kind of our staging server atm

Gonna close this PR in place of the one we have their.

**Pull request closed without merging**